### PR TITLE
Add build output spec

### PR DESCRIPTION
**Pull request by yufeih**
#2765 

@vwxyzh @vicancy @fenxu @OsmondJiang @superyyrrzz @herohua

### CODE REVIEW

File: docs/designs/build-output.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+- **Static build output xcopy deployable**: should just xcopy static build output to any static hosting server.
+
+
+## URL Schema
+
+An **URL** is an universal identifier that confirms to [URL Standard](https://url.spec.whatwg.org/).
+
+For *dynamic rendering*, `docfx` produces an output for lookup by the following URL schema:
+
+`https://{host}/{locale?}/{site-url}/?view={moniker?}`
```
Comment (OsmondJiang) [inline docs/designs/build-output.md]: why static and dynamic outputs are different?

Comment (yufeih) [reply docs/designs/build-output.md]: cannot have query string for static rendering. dynamic render has to follow docs.

File: docs/designs/build-output.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+```
+
+> `?` means optional.
+> When a docset is not localized, we will not create the `{locale}` folder.
+> This allows static websites with minimal url when localized and verioning is not needed.
+
+## File Outputs
+
+Each input file transforms to zero or more output files depending on its content type. `docfx` places output files in `output-path` relative to output directory. Regardless of static rendering or dynamic rendering, output path shares the same schema:
+
```
Comment (OsmondJiang) [inline docs/designs/build-output.md]: this is important, mark it strong

File: docs/designs/build-output.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+`{output-dir}/{locale?}/{moniker?}/{site-path}`
+
+```
+      locale    moniker                 site-path
+      |-^-| |------^------| |----------------^----------------|
+_site/en-us/netstandard-2.0/dotnet/api/system.string/index.html
```
Comment (superyyrrzz) [inline docs/designs/build-output.md]: >/index.html [](start = 52, length = 11)

make this as an option? keep the default output style as in v2

**Pull request merged**